t_fault_injection/test_fault_injection_base.py

```diff
@@ -84,26 +172,52 @@ def test(self, name, id, msg, retval):
         out = remote.client.run_cmd(cmd)
         # Specifies how many times failures may happen at most.
         # A value of -1 means “no limit”.
-        cmd = "printf %#x -1 > /sys/kernel/debug/fail_function/times"
+        cmd = f"printf %#x {times} > /sys/kernel/debug/fail_function/times"
         out = remote.client.run_cmd(cmd)
 
         client = self.get_client(id)
         request = client.create_request(method="GET", headers=[])
         client.start()
 
-        self.oops_ignore = ["ERROR"]
+        if should_fail:
+            self.oops_ignore = ["ERROR"]
         client.make_request(request)
-        self.assertFalse(client.wait_for_response(3))
-        self.assertTrue(client.wait_for_connection_close())
+        if should_fail:
+            self.assertFalse(client.wait_for_response(3))
+            self.assertTrue(client.wait_for_connection_close())
```

Comment:
Why do we need to wait for a response here?
I think `client.wait_for_response(3)` is not needed here. I suggest:
```suggestion
            self.assertTrue(client.wait_for_connection_close())
            self.assertIsNone(client.last_response)
```
We just check that the connection was closed and TempestaFW does not return a response

Reply:
Because we need to check warning message from tempesta. This is same as simple sleep (add comment)